flink-yarn/src/main/java/org/apache/flink/yarn/YarnApplicationFileUploader.java

```diff
@@ -335,7 +335,7 @@ public YarnLocalResourceDescriptor uploadFlinkDist(final Path localJarPath)
      *
      * @return list of class paths with the file name
      */
-    List<String> registerProvidedLocalResources() {
+    List<String> registerProvidedLocalResources(boolean shouldUsrLibExistInRemote) {
```

Comment:
Yes, I totally agree with the assumption and that it just have the same semantic with ship files. 
But in ship files, we will check if `usrlib` exists in ship files and if so and `INCLUDE_USER_JAR` is DISABLED, we will throw exception due to our previous discussion. 
The reason for me to adding such code is for this case: users have local `usrlib` and in the remote dirs specified by `provided.lib.dirs`, there is also a remote `usrlib`. I want to get rid of  such case as it could lead to overriding files in YARN cache's `usrlib`.

Reply:
I think no matter the `INCLUDE_USER_JAR` is configured, we should always not include `usrlib` in the user shipped files. This is also true for the provided lib.

Reply:
It makes sense to me. I will update the JIRA as well to emphasize  we should always not include `usrlib` in the user shipped files.   Codes are much easier with this.
Fixed in 986fff4543fbfd5016f087c53533b395af8c166f